compiler/rustc_interface/src/util.rs

```diff
@@ -194,27 +197,39 @@ pub(crate) fn run_in_thread_pool_with_globals<F: FnOnce() -> R + Send, R: Send>(
     let registry = sync::Registry::new(threads);
 
     if !sync::is_dyn_thread_safe() {
-        return run_in_thread_with_globals(edition, || {
+        return run_in_thread_with_globals(edition, |current_gcx| {
             // Register the thread for use with the `WorkerLocal` type.
             registry.register();
 
-            f()
+            f(current_gcx)
         });
     }
 
+    let current_gcx = CurrentGcx::new();
+    let current_gcx_ = FromDyn::from(current_gcx.clone());
+    let current_gcx = FromDyn::from(current_gcx);
```

Comment:
This naming scheme is really hard to read. There should be more than a single trailing underscore to distinguish two similar-but-different things. A comment would also be helpful to explain the distinction.

Reply:
They're exactly the same value. I've changed it to clone after the `FromDyn` construction to make that more clear.